### PR TITLE
116 Neutron weighting CL tool options

### PR DESCRIPTION
**Pull request by rebeccafair**
I think this is a lot cleaner this time around. It has been de-scoped so we will not include PDOS-type options for `intensity-map` yet, as to do this properly would require a `calculate_pdos_map` function that would have to return some kind of collection of 2D spectra, which hasn't been implemented yet. There are issues to add this in the future #158, #164.

The main changes are:
* Deprecated `--weights` in favour of `--weighting`
* Renamed `weighting=total` in `calculate_pdos` to `weighting=coherent-plus-incoherent`, verbose, but clear at least.
* Added `--weighting={coherent-dos,incoherent-dos,coherent-plus-incoherent-dos}` to `euphonic-dos`
* Added `--pdos` to `euphonic-dos`. `--pdos` with no arguments will plot PDOS grouped by species and total DOS (probably the most common use case?). You can also plot only specific species with something like `--pdos Si O`. In the future we could add things like `--pdos atoms`, `--pdos Si1 Si2` to plot per atom.
* Added `--weighting` as above to `euphonic-powder-map`
* Added `--pdos` to `euphonic-powder-map`, but it only allows a single argument in this case e.g. `--pdos Si`. This is to avoid having to implement multiple 2D subplots right now, this could be added later.
* Added `sample_sphere_pdos` function to facilitate the above 2 points. Its not currently being unit tested (only tested via `powder-map`) but that's no worse than the other `sample_sphere` functions.

This is the final big code change for PR #138. Once this is in I'll do a final wrap-up PR to update the user documentation and changelog.

### CODE REVIEW

**Review by ajjackson — APPROVED** [review]
This looks pretty thorough and systematic. I made a few small comments to look at re: edge-cases, names and docstrings, but no showstoppers.

File: euphonic/cli/powder_map.py [hunk 1]
```diff
@@ -100,13 +102,25 @@ def main(params: List[str] = None):
...
                 **calc_modes_kwargs)
-        elif args.weights == 'coherent':
+        elif 'dos' in args.weighting:
+            spectrum_1d_col = sample_sphere_pdos(
+                    fc, q,
+                    npts=npts, sampling=args.sampling, jitter=args.jitter,
+                    energy_bins=energy_bins,
+                    weighting=_get_pdos_weighting(args.weighting),
+                    **calc_modes_kwargs)
+            spectrum_1d = _get_output_dos(spectrum_1d_col, args.pdos)
```
[the remark this answers is not in the document]
Comment (rebeccafair) [reply euphonic/cli/powder_map.py]: Good point, renamed.

File: euphonic/cli/utils.py [hunk 2]
```diff
@@ -461,19 +519,62 @@ def _get_cli_parser(features: Collection[str] = {}
...
+        if 'pdos-weighting' in features:
+            _weighting_choices += _pdos_choices
+            desc += (', coherent neutron-weighted DOS, incoherent '
+                     'neutron-weighted DOS or total (coherent + incoherent) '
+                     'neutron-weighted DOS')
+        # We may have both 'pdos' and 'ins' so use separate if
+        if 'ins-weighting' in features:
+            _weighting_choices += _ins_choices
+            desc += ', coherent inelastic neutron scattering'
+        if 'ins-weighting' in features:
```
[the remark this answers is not in the document]
Comment (rebeccafair) [reply euphonic/cli/utils.py]: Good spot

File: euphonic/powder.py [hunk 3]
```diff
@@ -16,18 +17,17 @@ def sample_sphere_dos(fc: ForceConstants,
...
                       ) -> Spectrum1D:
-    """Sample the phonon DOS, averaging over a sphere of constant |q|
+    """
+    Calculate phonon DOS with QpointFrequencies.calculate_dos,
+    sampling over a sphere of constant |q|
 
     Parameters
     ----------
-
     fc
```
[the remark this answers is not in the document]
Comment (rebeccafair) [reply euphonic/powder.py:26]: The docstring seems to display fine without them, so to be consistent with other Euphonic docstrings we probably shouldn't have the spaces

**Comment by rebeccafair** [comment]
Ok, fixes done. CI tests have failed but only due to unrelated `mode_widths` calculations - this has been fixed in master by 38c185e and will be included once the feature branch has been rebased.

**Pull request merged**